src/misc_functions/highlight_util.ts

```diff
@@ -0,0 +1,20 @@
+import { SubAction } from "../types";
+
+export interface HighlightScope {
+    end: number;
+    scopes: string[];
+    start: number;
+}
+
+export function actionFromScopes(scopes: HighlightScope[]): SubAction[] {
+    return scopes.map<SubAction>(actionFromScope);
+}
+
+export function actionFromScope(scope: HighlightScope): SubAction {
+    return {
+        data: scope,
+        high: scope.end,
```

Comment:
If start and end are included in the SubAction, why are they in HighlightScope. Wouldn't it be better for this function to be something like:

```ts
export function createHighlight(scopes: string[], // (OR HighlightInnner if we want to support additional properties)
scopes:string[]
```